pkg/specgen/podspecgen.go

```diff
@@ -173,6 +173,7 @@ type PodSpecGenerator struct {
 	PodNetworkConfig
 	PodCgroupConfig
 	PodResourceConfig
+	InfraContainerSpec *SpecGenerator
```

Comment:
You should remove everything from the PodSpecGenerator struct which can be set in the InfraContainerSpec instead, e.g. the settings from PodNetworkConfig should be set in InfraContainerSpec. Right now I could set different settings in the PodNetworkConfig and the infra container. IMO all settings which only apply to the infra container should only be set in the InfraContainerSpec and not on the pod spec. Otherwise what happens if someone sets conflicting settings in the pod and infra spec, which one is used?

Reply:
Good idea, a simple way of doing this would be to remove the json tags from the PodNetworkConfig.

Reply:
So to be clear, if I remove the JSON tags that takes care of your concern right? Alot of the other duplicated spec stuff (I think) is useful for clarity in `pod inspect` and gets passed to the infra container properly.

Reply:
No I want you to completely remove the fields.

Reply:
Okay @Luap99 remove the network fields? is that what you mean

Reply:
To remove InfraContainerSpec from the json add this: ``json:"-"`` .This will tell the json marhsal/unmarshal to ignore this field.